src/node/rpc/common_endpoint_registry.h

```diff
@@ -3,13 +3,52 @@
 #pragma once
 
 #include "base_endpoint_registry.h"
+#include "ds/nonstd.h"
+#include "enclave/node_context.h"
 #include "http/http_consts.h"
 #include "http/ws_consts.h"
 #include "json_handler.h"
 #include "node/code_id.h"
+#include "node/historical_queries_adapter.h"
 
 namespace ccf
 {
+  static inline std::optional<ccf::TxID> txid_from_query_string(
+    EndpointContext& args)
+  {
+    const std::string prefix("transaction_id=");
+    const auto& path_params = args.rpc_ctx->get_request_query();
+
+    if (!nonstd::starts_with(path_params, prefix))
+    {
+      args.rpc_ctx->set_error(
+        HTTP_STATUS_BAD_REQUEST,
+        ccf::errors::InvalidInput,
+        "Query string must contain a single \'transaction_id\' parameter");
```

Comment:
Probably a stupid question: Why do we need the escapes on these single quotes?

Reply:
We don't.